### DISCUSSION TITLE
Editing PE (exe) in rust.

### DISCUSSION THREAD

**Discussion by Ars-Mond** [lief-project/LIEF#1099] [Q&A] [answered]
Good afternoon,
I am working on my project and I needed to create in my program the ability to edit a PE (exe) file. Specifically changing the size of a section and changing the value. The program is written in “rust” and I would not want to resort to clutter. Earlier I used “LIEF” in C++, I wanted to use it now. But I noticed that there is a “rust” binding.
How can I change a section in “PE” to “rust” using “LIEF” if it is possible?
Can i see some example?

p.s. I didn't find a save method in “pe::Binary”. And it's not clear on assigning a value at all.

Here is ~my old~ cpp code:
```cpp
int main() {
    std::unique_ptr<LIEF::PE::Binary> binary = LIEF::PE::Parser::parse("input.exe");

    auto& sections = binary->sections();
    if (!sections.empty()) {
        sections[0].size(0x2000);
        sections[0].content(std::vector<uint8_t>{0x90, 0x90, 0x90});
    }

    binary->write("output.exe");

    return 0;
}
```

**Comment by romainthomas** [comment, marked as answer]
Hi @Ars-Mond 

Right now the Rust bindings only expose "read-only" API so there is no way to modify a PE in Rust (at the moment".

It should be supported in the future though

**Reply by Ars-Mond** [reply]
Well, thank you.
